core/src/commonMain/kotlin/com/ustadmobile/core/domain/validateusername/ValidateUsernameUseCase.kt

```diff
@@ -0,0 +1,30 @@
+package com.ustadmobile.core.domain.validateusername
+
+
+class ValidateUsernameUseCase {
+
+    /**
+     * Validates a username according to the following rules:
+     * - Must not contain spaces, tabs, or banned special characters like !@#$%^&*()
+     * - Non-English Unicode characters are allowed
+     * - Must be 3–15 characters long after trimming
+     *
+     * @param username The username string to validate
+     * @return The valid username (trimmed) if valid, or null if invalid
```

Comment:
If banned characters are found, then this should remove/replace them with and return whatever is valid 

add an argument: ```invalidReplacement: String = ""```

Any banned characters should then be replaced with the replacement. This is going to be required when running the migration to fix any invalid usernames found.